File: MiniDOMBase/DMD/Sistema/Calendar/Providers/ICalendarProvider.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica

Partial Class Sistema

    ''' <summary>
    ''' Classe base del provider di attività del calendario
    ''' </summary>
    ''' <remarks></remarks>
    Public Interface ICalendarProvider

        ''' <summary>
        ''' Crea un nuovo oggetto 
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function InstantiateNewItem() As Object


        Function GetCreateCommand() As String

        ''' <summary>
        ''' Restituisce un testo breve che descrive il tipo di provider
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetShortDescription() As String

        ''' <summary>
        ''' Restituisce un array di tipi di oggetto supportati da questo provider
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetSupportedTypes() As System.Type()

        ''' <summary>
        ''' Restituisce le persone con eventi gestiti da questo provider nell'intervallo di date specificato
        ''' </summary>
        ''' <param name="fromDate"></param>
        ''' <param name="toDate"></param>
        ''' <param name="ufficio"></param>
        ''' <param name="operatore"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetActivePersons(ByVal nomeLista As String, ByVal fromDate As Date?, ByVal toDate As Date?, Optional ByVal ufficio As Integer = 0, Optional ByVal operatore As Integer = 0) As CCollection(Of CActivePerson)

        ''' <summary>
        ''' Restituisce le scadenza gestite da questo provider
        ''' </summary>
        ''' <param name="fromDate"></param>
        ''' <param name="toDate"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetScadenze(ByVal fromDate As Date?, ByVal toDate As Date?) As CCollection(Of ICalendarActivity)

        ''' <summary>
        ''' Restituisce le aziende pendenti
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetPendingActivities() As CCollection(Of ICalendarActivity)

        ''' <summary>
        ''' Restituisce l'elenco delle cose da fare per l'utente specificato
        ''' </summary>
        ''' <param name="user"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetToDoList(ByVal user As CUser) As CCollection(Of ICalendarActivity)

        ''' <summary>
        ''' Restituisce il nome che identifica univocamente il provider all'interno del sistema
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        ReadOnly Property UniqueName As String


        Sub SaveActivity(ByVal item As ICalendarActivity, Optional ByVal force As Boolean = False)

        Sub DeleteActivity(ByVal item As ICalendarActivity, Optional ByVal force As Boolean = False)

    End Interface


End Class

File: MiniDOMBase/DMD/Sistema/Calendar/Providers/Iinterface/ICalendarProvider.cs
using System;
using System.Collections;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using static minidom.Sistema;
using static minidom.Anagrafica;
using minidom.repositories;
using System.Collections.Generic;


namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Interfaccia implementata dai provider di attività del calendario
        /// </summary>
        /// <remarks></remarks>
        public interface ICalendarProvider
        {

            /// <summary>
            /// Crea un nuovo oggetto
            /// </summary>
            /// <returns></returns>
            /// <remarks></remarks>
            object InstantiateNewItem();

            /// <summary>
            /// Restituisce la stringa che il provider identifica come stringa di creazione
            /// </summary>
            /// <returns></returns>
            string GetCreateCommand();

            /// <summary>
            /// Restituisce un testo breve che descrive il tipo di provider
            /// </summary>
            /// <returns></returns>
            /// <remarks></remarks>
            string GetShortDescription();

            /// <summary>
            /// Restituisce un array di tipi di oggetto supportati da questo provider
            /// </summary>
            /// <returns></returns>
            /// <remarks></remarks>
            Type[] GetSupportedTypes();

            /// <summary>
            /// Restituisce le persone con eventi gestiti da questo provider nell'intervallo di date specificato
            /// </summary>
            /// <param name="fromDate"></param>
            /// <param name="toDate"></param>
            /// <param name="ufficio"></param>
            /// <param name="operatore"></param>
            /// <param name="nomeLista"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            CCollection<CActivePerson> GetActivePersons(string nomeLista, DateTime? fromDate, DateTime? toDate, int ufficio = 0, int operatore = 0);

            /// <summary>
            /// Restituisce le scadenza gestite da questo provider
            /// </summary>
            /// <param name="fromDate"></param>
            /// <param name="toDate"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            CCollection<ICalendarActivity> GetScadenze(DateTime? fromDate, DateTime? toDate);

            /// <summary>
            /// Restituisce le aziende pendenti
            /// </summary>
            /// <returns></returns>
            /// <remarks></remarks>
            CCollection<ICalendarActivity> GetPendingActivities();

            /// <summary>
            /// Restituisce l'elenco delle cose da fare per l'utente specificato
            /// </summary>
            /// <param name="user"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            CCollection<ICalendarActivity> GetToDoList(CUser user);

            /// <summary>
            /// Restituisce il nome che identifica univocamente il provider all'interno del sistema
            /// </summary>
            /// <value></value>
            /// <returns></returns>
            /// <remarks></remarks>
            string UniqueName { get; }

            /// <summary>
            /// Salva l'attività
            /// </summary>
            /// <param name="item"></param>
            /// <param name="force"></param>
            void SaveActivity(ICalendarActivity item, bool force = false);

            /// <summary>
            /// Elimina l'attività
            /// </summary>
            /// <param name="item"></param>
            /// <param name="force"></param>
            void DeleteActivity(ICalendarActivity item, bool force = false);
        }
    }
}